components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/ClaimUtil.java

```diff
@@ -83,13 +83,16 @@ private ClaimUtil() {
 
             UserStoreManager userstore = realm.getUserStoreManager();
 
-            TokenMgtDAO tokenMgtDAO = new TokenMgtDAO();
-            AccessTokenDO accessTokenDO = tokenMgtDAO.retrieveAccessToken(tokenResponse.getAuthorizationContextToken()
-                    .getTokenString(), false);
+            AccessTokenDO accessTokenDO = OAuth2Util.getAccessTokenDOfromTokenIdentifier(tokenResponse
+                    .getAuthorizationContextToken().getTokenString());
             ApplicationManagementService applicationMgtService = OAuth2ServiceComponentHolder.getApplicationMgtService();
-            String clientId = null;
+            String clientId;
             if (accessTokenDO != null) {
                 clientId = accessTokenDO.getConsumerKey();
+            } else {
+                // this means the token is not active so we can't proceed further
+                throw new UserInfoEndpointException(OAuthError.ResourceResponse.INVALID_TOKEN,
+                        "Invalid Access Token. Access token is not ACTIVE.");
```

Comment:
@ruwanta at this point the only way this exception could occur is if the token is in the database but not in the ACTIVE state. If there was no AccessToken corresponding to a particular string xyz in the DB we wouldn't have come up to this point. ie. it would have failed at this [check](https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/OpenIDConnectUserEndpoint.java#L70).